Support/bin/latextidy.pl

```diff
@@ -197,10 +197,22 @@
 }
 
 # At this point each \label appears on a line of its own.
-# This hack is designed to restore \labels next to \begin, \end and suchlike, so that code folding leaves the \label visible.
-$string =~ s/({[^}]*}[ \t]*)(\n[ \t]*)((\[[^\]]*\][ \t]*)*(\\label[ \t]*(\[[^\]]*\][ \t]*)*{[^}]*}[ \t]*)*)\n?[ \t]*/$1$3$2/mg;
-$string =~ s/((\\label[ \t]*(\[[^\]]*\][ \t]*)*{[^}]*}[ \t]*)+)(\n[ \t]*)(\\end{[^}]*})/$4$1$5/mg;
-$string =~ s/(\\section.*(\\label[ \t]*(\[[^\]]*\][ \t]*)*{[^}]*}[ \t]*))(\S)/$1\n$4/mg;
+# This hack is designed to restore \labels next to \begin, \end and suchlike
+# so that code folding leaves the \label visible.
+my $option = '(\[[^\]]*\][ \t]*)';
+my $arg = '({[^}]*}[ \t]*)';
+$string =~ s/(?<argument>${arg})
+		(?<newline_then_indent>\n[ \t]*)
+		(?<options_then_labels>${option}*(\\label[ \t]*${option}?${arg})+)
+		\n?[ \t]*
+	/$+{argument}$+{options_then_labels}$+{newline_then_indent}/mgx;
+$string =~ s/(?<labels>(\\label[ \t]*${option}?${arg})+)
+		(?<newline_then_indent>\n[ \t]*)
+		(?<end_argument>\\end{[^}]*})
+	/$+{newline_then_indent}$+{labels}$+{end_argument}/mgx;
+$string =~ s/(?<labelled_section>\\section.*(\\label[ \t]*${option}?${arg})+)
+		(?<paragraph>\S)
+	/$+{labelled_section}\n$+{paragraph}/mgx;
 $string =~ s/(?<=\S)[ \t]*(?:\n[ \t]*){0,2}(\\section)/\n\n\n$1/mg;
```

Comment:
Could you maybe move the code from this and the next line (line 216–217) into a separate commit? I think this would make sense, since this code does something different  – add empty lines before sections –, than the code in the lines 200–215. To do that you can use git's [interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).

By the way: If you want to improve your commit messages, [this tutorial](http://chris.beams.io/posts/git-commit) shows the conventions used by most projects. I think it would also make sense to start the commit messages with something like “Tidy:…”, since then other users immediately see which command you improved.